### ISSUE TITLE
Hoon parser crashes on bad ~0... jam literal

### ISSUE THREAD

**Issue by ohAitch** [urbit/urbit#1473]
If you try to type `~00` into dojo, this causes an unlink-relink:

```
[%dojo-peer-replaced 2]
[unlinked from [p=~zod q=%dojo]]
/~zod/home/0/app/dojo:<[1.137 24].[1.138 21]>
[%poke [%det ler=[own=50 his=93] haw=0v6.bu38e.10dfq.m8tvv.qjhbt.rknje ted=[%mor p=~[[%ins p=2 q=~-0]]]]]
/~zod/home/0/app/dojo:<[1.137 38].[1.138 21]>
/~zod/home/0/app/dojo:<[1.123 3].[1.124 22]>
/~zod/home/0/app/dojo:<[1.124 3].[1.124 22]>
/~zod/home/0/app/dojo:<[1.124 11].[1.124 22]>
/~zod/home/0/app/dojo:<[993 5].[1.000 7]>
/~zod/home/0/app/dojo:<[994 5].[1.000 7]>
/~zod/home/0/app/dojo:<[996 5].[1.000 7]>
/~zod/home/0/app/dojo:<[997 13].[997 28]>
/~zod/home/0/app/dojo:<[946 5].[958 27]>
/~zod/home/0/app/dojo:<[949 5].[958 27]>
/~zod/home/0/app/dojo:<[950 5].[958 27]>
/~zod/home/0/app/dojo:<[954 5].[958 27]>
/~zod/home/0/app/dojo:<[954 9].[954 37]>
/~zod/home/0/app/dojo:<[954 13].[954 37]>
/~zod/home/0/app/dojo:<[799 5].[807 31]>
/~zod/home/0/app/dojo:<[800 5].[807 31]>
/~zod/home/0/app/dojo:<[801 5].[807 31]>
/~zod/home/0/app/dojo:<[802 5].[807 31]>
/~zod/home/0/app/dojo:<[802 9].[802 50]>
/~zod/home/0/app/dojo:<[802 13].[802 50]>
[%drum-coup-fail ~zod 1 p=~zod q=%dojo]
[linked to [p=~zod q=%dojo]]
```

The `%blob` coin form is not really used for anything serious, so we could probably get away with virtualizing all `cue` calls in `twid:so`?

**Comment by joemfb** [comment]
Fixed in #3638

**Comment by vvisigoth** [comment]
wow
—
~poldec-tonteg
http://urbit.org
